src/compiler/utilities.ts

```diff
@@ -2414,6 +2414,27 @@ namespace ts {
         return output;
     }
 
+    export const stringify: (value: any) => string = JSON && JSON.stringify ? JSON.stringify : function stringify(value: any): string {

```

Comment:
The fallback will loop forever (or, rather, run out of stack space: `RangeError: Maximum call stack size exceeded` in node) on a circularly referential object structure. That should probably get caught before it happens.